pkg/providers/cloudstack/cloudstack.go

```diff
@@ -782,9 +784,10 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, datacenterConfigSpec v1alpha1
 
 		// Add no-proxy defaults
 		noProxyList = append(noProxyList, clusterapi.NoProxyDefaults()...)
-		cloudStackManagementApiEndpointHostname, err := getHostnameFromUrl(datacenterConfigSpec.ManagementApiEndpoint)
-		if err == nil {
-			noProxyList = append(noProxyList, cloudStackManagementApiEndpointHostname)
+		for _, az := range datacenterConfigSpec.AvailabilityZones {
```

Comment:
Can we have one single proxy setup method which fills all the proxy configs? it can be used in both `buildTemplateMapCP` and `buildTemplateMapMD`. Looks like a lot of duplicate code here: line 788 vs 875